wp-content/themes/pub/wporg-learn-2024/src/upcoming-online-workshops/block.php

```diff
@@ -0,0 +1,83 @@
+<?php
+namespace WordPressdotorg\Theme\Learn_2024\Upcoming_Online_Workshops;
+
+use DateTime;
+use function WPOrg_Learn\Events\{get_discussion_events};
+
+add_action( 'init', __NAMESPACE__ . '\init' );
+
+/**
+ * Registers the block using the metadata loaded from the `block.json` file.
+ * Behind the scenes, it registers also all assets so they can be enqueued
+ * through the block editor in the corresponding context.
+ *
+ * @see https://developer.wordpress.org/reference/functions/register_block_type/
+ */
+function init() {
+	register_block_type(
+		dirname( dirname( __DIR__ ) ) . '/build/upcoming-online-workshops',
+		array(
+			'render_callback' => __NAMESPACE__ . '\render',
+		)
+	);
+}
+
+/**
+ * Render the block content.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ *
+ * @return string Returns the block markup.
+ */
+function render( $attributes, $content, $block ) {
+	$upcoming_online_workshops = get_discussion_events();
+
+	if ( empty( $upcoming_online_workshops ) || is_wp_error( $upcoming_online_workshops ) ) {
+		$content = '<!-- wp:group {"layout":{"type":"constrained"}} -->
+		<div class="wp-block-group"><!-- wp:paragraph -->
+		<p>' . __( 'No upcoming online workshops found.', 'wporg-learn' ) . '</p>
+		<!-- /wp:paragraph --></div>
+		<!-- /wp:group -->';
+	} else {
+		$content = '<!-- wp:group {"style":{"spacing":{"blockGap":"var:preset|spacing|50"}},"className":"is-style-cards-grid","layout":{"type":"grid","columnCount":"3","minimumColumnWidth":null}} --><div class="wp-block-group is-style-cards-grid">';
+
+		foreach ( $upcoming_online_workshops as $workshop ) {
+			$timestamp = strtotime( $workshop['date_utc'] ) - (int) $workshop['date_utc_offset'];
+
+			$content .= sprintf(
+				'<!-- wp:wporg/link-wrapper -->
+				<a class="wp-block-wporg-link-wrapper" href="%1$s">
```

Comment:
This will need to change to a layout like the other sections when the featured image is supported in the Meeting plugin. For now it uses a simple card link wrapper like we use on other home pages, like Developer Resources.